Property `ref` does not exist on `<Link>` component when `typedRoutes: true`

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 18.16.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.8-canary.6
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.9.5
    Next.js Config:
      output: N/A
```


### Which area(s) of Next.js are affected? (leave empty if unsure)

Routing (next/router, next/navigation, next/link), TypeScript (plugin, built-in types)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/mateuszaliyev/next-typed-routes-link-ref

### To Reproduce

1. Clone the repository
2. Install dependencies (`npm install`)
3. Run development server so TypeScript plugin can generate types (`npm run dev`)
4. Run `npx tsc` or check the `/app/page.tsx` file to see the type error

### Describe the Bug

Property `ref` does not exist on `<Link>` component when `experimental.typedRoutes = true`.

```
$ npx tsc
app/page.tsx:10:20 - error TS2322: Type '{ children: string; href: "/"; ref: MutableRefObject<HTMLAnchorElement>; }' is not assignable to type 'IntrinsicAttributes & LinkRestProps & { href: UrlObject | RouteImpl<"/">; }'.
  Property 'ref' does not exist on type 'IntrinsicAttributes & LinkRestProps & { href: UrlObject | RouteImpl<"/">; }'. Did you mean 'href'?

10     <Link href="/" ref={ref}>
                      ~~~


Found 1 error in app/page.tsx:10
```

![image](https://github.com/vercel/next.js/assets/82393815/68a72b80-65a1-418b-9dda-97693b825e56)


### Expected Behavior

Property `ref` should exist on `<Link>` component regardless of the `experimental.typedRoutes` setting.

### Which browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

--- a/packages/next/src/build/webpack/plugins/next-types-plugin.ts
+++ b/packages/next/src/build/webpack/plugins/next-types-plugin.ts
@@ -413,7 +413,7 @@ declare module 'next' {
 
 declare module 'next/link' {
   import type { LinkProps as OriginalLinkProps } from 'next/dist/client/link'
-  import type { AnchorHTMLAttributes, DetailedHTMLAttributes } from 'react'
+  import type { AnchorHTMLAttributes, DetailedHTMLProps } from 'react'
   import type { UrlObject } from 'url'
   
   type LinkRestProps = Omit<

--- a/packages/next/src/build/webpack/plugins/next-types-plugin.ts
+++ b/packages/next/src/build/webpack/plugins/next-types-plugin.ts
@@ -413,11 +413,17 @@ declare module 'next' {
 
 declare module 'next/link' {
   import type { LinkProps as OriginalLinkProps } from 'next/dist/client/link'
-  import type { AnchorHTMLAttributes } from 'react'
+  import type { AnchorHTMLAttributes, DetailedHTMLAttributes } from 'react'
   import type { UrlObject } from 'url'
   
   type LinkRestProps = Omit<
-    Omit<AnchorHTMLAttributes<HTMLAnchorElement>, keyof OriginalLinkProps> &
+    Omit<
+      DetailedHTMLProps<
+        AnchorHTMLAttributes<HTMLAnchorElement>,
+        HTMLAnchorElement
+      >,
+      keyof OriginalLinkProps
+    > &
       OriginalLinkProps,
     'href'
   >

